### DISCUSSION TITLE
Adding query packs via cli

### DISCUSSION THREAD

**Discussion by Wilk4013** [Security-Onion-Solutions/securityonion#4603] [Unsupported Versions] [answered]
Hello,
Is it possible add Osquery query packs into Fleet DM via the cli or would I just have to add individual queries via the Fleet DM UI and create a query pack from there? I see the default query packs look like they are in the /opt/so/conf/fleet/packs/palantir/Fleet/Endpoint/packs. I placed my .yml file there but it doesn't show up in the UI.

Thanks

**Comment by defensivedepth** [comment, marked as answer]
@Wilk4013 You can add them via fleetctl, a cli tool provided by the maintainers of FleetDM. Docs for that can be found here: https://github.com/fleetdm/fleet/blob/main/docs/1-Using-Fleet/2-fleetctl-CLI.md

fleetctl is in the FleetDM container, you can run it like so (It may require you to login):

`sudo docker exec -it so-fleet fleetctl `

The following directory is mapped to the container, so you can drop your query packs in there and reference it: 
/opt/so/conf/fleet/packs

For instance:

`sudo docker exec -it so-fleet fleetctl apply -f  /packs/<yourpack>.yaml`

**Reply by Wilk4013** [reply]
Thank you Josh!!

**Comment by itpropaul** [comment]
@defensivedepth 
Is there a one-liner or utility that can import all of the packs that ship with Security Onion at once instead of doing them one by one?

cc @searsb

**Reply by defensivedepth** [reply]
Here is how we do it during the FleetDM setup process:

https://github.com/Security-Onion-Solutions/securityonion/blob/master/salt/common/tools/sbin/so-fleet-setup#L34

**Reply by itpropaul** [reply]
Thanks @defensivedepth
